src/Core/Geographic/GeoCoordinate.js

```diff
@@ -18,60 +21,73 @@ export const UNIT = {
-{
-    unit = unit || UNIT.RADIAN;
-
-    if (unit === UNIT.RADIAN) {
-        coord[id] = value;
-    } else if (unit === UNIT.DEGREE) {
-        coord[id] = mE.degToRad(value);
+// Only support explicit conversions
+function convert(inUnit, outUnit, value) {
+    if (inUnit == outUnit || outUnit === undefined) {
+        return value;
+    } else {
+        switch (inUnit) {
+            case UNIT.RADIAN: {
+                if (outUnit === UNIT.DEGREE) {
+                    return mE.radToDeg(value);
+                } else {
+                    // from meter require CRS and projection etc
+                    throw new Error('Cannot convert from Meters to Radians');
+                }
+            }
+            case UNIT.DEGREE: {
+                if (outUnit === UNIT.RADIAN) {
+                    return mE.degToRad(value);
+                } else {
+                    // from meter require CRS and projection etc
+                    throw new Error('Cannot convert from Meters to Degree');
+                }
+            }
+            default:
+                throw new Error(`Cannot convert from unit ${inUnit} to unit ${outUnit}`);
+        }
     }
-    return coord[id];
-};
-
-var setCoordinate = function setCoordinate(coordinate, longitude, latitude, altitude, unit) {
-    unit = unit || UNIT.RADIAN;
-
-    setCoordinateValue(unit, coordinate, COORD.LONG, longitude);
-    setCoordinateValue(unit, coordinate, COORD.LAT, latitude);
+}
 
+var setCoordinate = function setCoordinate(coordinate, longitude, latitude, altitude, inUnit, toUnit) {
+    coordinate[COORD.LONG] = convert(inUnit, toUnit, longitude);
+    coordinate[COORD.LAT] = convert(inUnit, toUnit, latitude);
     coordinate[COORD.ALT] = altitude;
 };
 
 function GeoCoordinate(longitude, latitude, altitude, unit) {
+    if (longitude && unit === undefined) {
+        throw new Error('Cannot build a GeoCoordinate without a unit');
+    }
+
+    this.unit = unit;
     this.coordinate = new Float64Array(3);
 
-    setCoordinate(this.coordinate, longitude, latitude, altitude, unit);
+    setCoordinate(this.coordinate, longitude, latitude, altitude, unit, unit);
 }
 
 GeoCoordinate.prototype.constructor = GeoCoordinate;
 
 GeoCoordinate.prototype.longitude = function longitude(unit) {
-    return getCoordinateValue(unit, this.coordinate, COORD.LONG);
+    return convert(this.unit, unit, this.coordinate[COORD.LONG]);
 };
 
+GeoCoordinate.prototype.x = GeoCoordinate.prototype.longitude;
+
 GeoCoordinate.prototype.setLongitude = function setLongitude(longitude, unit) {
-    setCoordinateValue(unit, this.coordinate, COORD.LONG, longitude);
+    this.coordinate[COORD.LONG] = convert(unit, this.unit, longitude);
+    return this;
 };
 
 GeoCoordinate.prototype.latitude = function latitude(unit) {
-    return getCoordinateValue(unit, this.coordinate, COORD.LAT);
+    return convert(this.unit, unit, this.coordinate[COORD.LAT]);
```

Comment:
This comment was refering to these methods: `c.longitude()` which would return longitude in the *current unit*, but this current unit might be unknown to the caller.

I prefer `c.longitude(my_unit)` where the caller control the unit of the result.